### PR TITLE
[tests] enhance Cert_5_1_06_RemoveRouterId

### PR DESCRIPTION
**Pull request by wgtdkp**
1. verify messages' format;
2. remove pass criteria not required by R67 [5.1.6];

### CODE REVIEW

File: tests/scripts/thread-cert/Cert_5_1_06_RemoveRouterId.py [hunk 1]
```diff
@@ -100,27 +100,32 @@ def test(self):
...
         msg.assertCoapMessageRequestUriPath("/a/as")
 
-        msg = leader_messages.next_coap_message("2.04")
+        leader_messages.next_coap_message("2.04")
 
         # 3 - Router1
-        router1_messages.next_mle_message(mle.CommandType.PARENT_REQUEST)
-        leader_messages.next_mle_message(mle.CommandType.PARENT_RESPONSE)
+        msg = router1_messages.next_mle_message(mle.CommandType.PARENT_REQUEST)
+        msg.assertSentWithHopLimit(255)
```
Comment (librasungirl) [inline tests/scripts/thread-cert/Cert_5_1_06_RemoveRouterId.py]: use existing `check_parent_request()` function?
https://github.com/openthread/openthread/blob/master/tests/scripts/thread-cert/command.py#L237

one suggestion is to add parameter for `check_parent_request()` to indicate whether this is first parent request attempt (scan mask should be 0x80) or second attempt (scanmask 0xc0).

File: tests/scripts/thread-cert/Cert_5_1_06_RemoveRouterId.py [hunk 2]
```diff
@@ -100,27 +100,32 @@ def test(self):
...
+        msg.assertMleMessageContainsTlv(mle.TlvRequest)
+        msg.assertMleMessageDoesNotContainTlv(mle.AddressRegistration)
 
-        router1_messages.next_mle_message(mle.CommandType.CHILD_ID_REQUEST)
-        msg = leader_messages.next_mle_message(mle.CommandType.CHILD_ID_RESPONSE)
-        msg.assertSentToNode(self.nodes[ROUTER1])
-            
         msg = router1_messages.next_coap_message(code="0.02", uri_path="/a/as")
         msg.assertCoapMessageContainsTlv(network_layer.MacExtendedAddress)
         msg.assertCoapMessageContainsTlv(network_layer.Status)
```
Comment (librasungirl) [inline tests/scripts/thread-cert/Cert_5_1_06_RemoveRouterId.py]: add one common `check_address_solicit()` function? like `check_address_release()` in command.py

In this scenario, there should be RLOC16 TLV in address solicit, suggest to use one parameter to indicate whether RLOC16 TLV should be included.

File: tests/scripts/thread-cert/Cert_5_1_06_RemoveRouterId.py [hunk 3]
```diff
@@ -100,27 +100,32 @@ def test(self):
...
+        msg.assertSentWithHopLimit(255)
+        msg.assertSentToDestinationAddress("ff02::2")
+        msg.assertMleMessageContainsTlv(mle.Mode)
+        msg.assertMleMessageContainsTlv(mle.Challenge)
+        msg.assertMleMessageContainsTlv(mle.ScanMask)
+        msg.assertMleMessageContainsTlv(mle.Version)
+
+        msg = router1_messages.next_mle_message(mle.CommandType.CHILD_ID_REQUEST)
+        msg.assertSentToNode(self.nodes[LEADER])
+        msg.assertMleMessageContainsTlv(mle.Response)
```
Comment (librasungirl) [inline tests/scripts/thread-cert/Cert_5_1_06_RemoveRouterId.py]: similarly reuse existing `check_child_id_request()`?

File: tests/scripts/thread-cert/Cert_5_1_06_RemoveRouterId.py (hunk 1 above)
Comment (wgtdkp) [reply tests/scripts/thread-cert/Cert_5_1_06_RemoveRouterId.py]: done. 
The test plan document requires only checking if scanMakk TLV presents in the message,  so doesn't add this parameter.

**Pull request merged**